src/common/snippets/src/pass/softmax_reshape_elimination.cpp

```diff
@@ -24,12 +25,16 @@ ov::snippets::pass::SoftmaxReshapeElimination::SoftmaxReshapeElimination() {
             auto softmax = pattern_to_output[m_softmax].get_node_shared_ptr();
             auto reshape1 = pattern_to_output[m_reshape1].get_node_shared_ptr();
 
+            if (reshape0->get_output_target_inputs(0).size() != 1)
+                return false;
```

Comment:
2 minor points:
1. Consumers count check could be done in matcher (ov::pass::pattern::consumers_count(1) as a predicate)
2. Formally, Softmax must have only 1 consumer as well

Reply:
Done, thanks